tests/test_inputmedia.py

```diff
@@ -443,6 +443,64 @@ async def test_send_media_group_photo(self, bot, chat_id, media_group):
             mes.caption_entities == [MessageEntity(MessageEntity.BOLD, 0, 5)] for mes in messages
         )
 
+    async def test_send_media_group_throws_error_with_group_caption_and_individual_captions(
+        self, bot, chat_id, media_group
+    ):
+        with pytest.raises(
+            ValueError, match="You can only supply either group caption or media with captions."
+        ):
+            await bot.send_media_group(chat_id, media_group, caption="foo")
+
+    @pytest.mark.parametrize(
+        "caption, parse_mode, caption_entities",
+        [
+            # same combinations of caption options as in media_group fixture
+            ("*photo* 1", "Markdown", None),
+            ("<b>photo</b> 1", "HTML", None),
+            ("photo 1", None, [MessageEntity(MessageEntity.BOLD, 0, 5)]),
+        ],
+    )
+    @flaky(3, 1)
+    async def test_send_media_group_with_group_caption(
+        self,
+        bot,
+        chat_id,
+        media_group,
+        caption,
+        parse_mode,
+        caption_entities,
+    ):
+        # clear media_group of all_captions
```

Comment:
we should also check that the `media_group` is not altered. to do this please create a copy `media_group` (e.g. `copied_media_group = media_group.copy()`). Then after calling `send_media_group` add assertions for `copied_media_group == media_group` and `media_group[0] is copied_media_group[0]` (which double-ensures that the first item wasn't replaced)

Reply:
Absolutely